### PR TITLE
Frontend implementation

### PR DESCRIPTION
**Pull request by bfokss**
**Finished basic frontend implementation.**
You can check if "authentication" works.
To test each user's page just pass his id in URL like so:
localhost/users?id=2

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,26 +1,52 @@
-from flask import render_template
+from flask import render_template, request, redirect, url_for
 import pandas as pd
 
+
 from project import app, db
 from project.models import User, Card, Event
 import project.api
```
Comment (wdebsqi) [inline app.py]: This import is redundant but I'll remove it in another PR, where we'll get rid of api.py file.

File: app.py [hunk 2]
```diff
@@ -1,26 +1,52 @@
...
+        ).select_from(Event).join(Card, full=True).join(User, full=True).filter(User.id==userId).order_by(Event.log_time.desc())
+
+        data = query.all()
+        userFirstRow = query.first()
+        userData = pd.DataFrame(data)
+
+        trainings = db.session.query(Event.log_time, Event.event_type, Card.user_id, User.first_name).select_from(Event).join(Card, full=True).join(User, full=True).filter(User.id==userId,Event.event_type=="start").order_by(Event.log_time.desc()).all()
+        userTrainings = pd.DataFrame(trainings)
+
+        userTrainingsLimit = 10
```
Comment (wdebsqi) [inline app.py]: Consider changing this to const and moving to the top of the file. It's not necessary, up to you.

File: app.py [hunk 3]
```diff
@@ -1,26 +1,52 @@
...
+def user_index():
+    userId = request.args.get('id')
+
+    userExists = db.session.query(User.id).filter_by(id=userId).first() is not None
+
+    
+    if ('id' not in request.args) or (not userExists):
+        return redirect(url_for('index'))
+    
+    else:
```
Comment (wdebsqi) [inline app.py]: FYI you don't have to add `else` here, since you already `return` above.

File: app.py [hunk 4]
```diff
@@ -1,26 +1,52 @@
...
+            User.id,
+            User.first_name,
+            User.last_name
+        ).select_from(Event).join(Card, full=True).join(User, full=True).filter(User.id==userId).order_by(Event.log_time.desc())
+
+        data = query.all()
+        userFirstRow = query.first()
+        userData = pd.DataFrame(data)
+
+        trainings = db.session.query(Event.log_time, Event.event_type, Card.user_id, User.first_name).select_from(Event).join(Card, full=True).join(User, full=True).filter(User.id==userId,Event.event_type=="start").order_by(Event.log_time.desc()).all()
```
Comment (wdebsqi) [inline app.py]: I don't this querying all the data here and cut it later in `sessions.html` file is a good idea. Instead consider using `.limit()` at the end of the query.

**Review by wdebsqi — CHANGES_REQUESTED** [review]
Everything looks really good. The only important thing I would recommend changing is line 40 in `app.py` (see the attached comment for more information). The rest is up to you :)

File: app.py (hunk 3 above)
Comment (bfokss) [reply app.py]: Else is used here to render another template if the condition in if is not met

File: app.py (hunk 2 above)
Comment (bfokss) [reply app.py]: Will do

File: app.py (hunk 3 above)
Comment (wdebsqi) [reply app.py]: Yeah, I know. Just FYI you don't have to use `else` in this case - if condition is true the script will `return`. Thus everything below line 23 will be ran only if condition is not true. But what you've written is not logically wrong - just a matter of personal preference.

Comment (bfokss) [reply app.py]: Okay now I understand

**Review by bfokss — COMMENTED** [review]
Resolved every issue

**Review by wdebsqi — APPROVED** [review]

**Pull request merged**